### ISSUE TITLE
301 redirects not handled correctly

### ISSUE THREAD

**Issue by wakeldakel** [square/okhttp#5910]
I have to work with the following setup:

1. `example.com/picture.jpg` redirects (HTTP 301) to `static.example.com/picture.jpg`
2. DNS A record of `static.example.com` subdomain is set to a different IP address than `example.com`, so 2 different machines

I have verified that this is working fine with all kinds of different clients.
When I use OkHttp to get `static.example.com/picture.jpg` directly, the expected result is returned. However, a request to `example.com/picture.jpg` gives me a `java.net.ProtocolException: Too many follow-up requests: 21`.

I'm only a few days into Android development, but this is what I could find so far:
followUp request responses in `RetryAndFollowUpInterceptor` all are correctly set to `static.example.com/picture.jpg`, but still contain a 301 response. This is impossible as the static server isn't even capable of 301 redirects. To me it feels like the request is sent to the same IP address even if the subdomain has changed (just a guess though).

Thank you in advance for every idea!

**Comment by swankjesse** [comment]
Do the domains share both IP addresses and TLS certificates?

**Comment by yschimke** [comment]
Real hosts and URLs would help a lot here. This is almost certainly server config issue. Also what does Firefox and Chrome see?

**Comment by wakeldakel** [comment]
Chrome, Firefox, Edge, IE, wget, curl and URLSession from the iOS SDK all handle the 301 with a single redirect followed by loading the file. No client I have tried had problems handling the redirect except OkHttp.

The domains use two different certificates and are pointed to two different IP addresses.

Example: `https://earth.sexy/facts/002/card.png` redirects to `https://files.earth.sexy/facts/002/card.png`.

This is the code I'm using. Unspectacular, but maybe it helps:
```
val client = OkHttpClient.Builder().build()
val response: Response = try {
    client.newCall(Request.Builder().url(URL("https://earth.sexy/facts/002/card.png")).build()).execute()
} catch (_: Throwable) { return }
```

**Comment by yschimke** [comment]
Slight tweak to the client provides useful output

```
  val listener = LoggingEventListener.Factory()
  val client = OkHttpClient.Builder().eventListenerFactory(listener).build()
```

```

Mar 29, 2020 11:48:45 AM okhttp3.internal.platform.Platform log
INFO: [82 ms] dnsStart: earth.sexy
Mar 29, 2020 11:48:45 AM okhttp3.internal.platform.Platform log
INFO: [159 ms] dnsEnd: [earth.sexy/18.194.112.90, earth.sexy/18.196.93.106]

Mar 29, 2020 11:48:45 AM okhttp3.internal.platform.Platform log
INFO: [495 ms] dnsStart: files.earth.sexy
Mar 29, 2020 11:48:45 AM okhttp3.internal.platform.Platform log
INFO: [536 ms] dnsEnd: [files.earth.sexy/13.224.77.59, files.earth.sexy/13.224.77.72, files.earth.sexy/13.224.77.18, files.earth.sexy/13.224.77.3]
```

**Comment by yschimke** [comment]
Are you running with 4.4.0? 4.4.1 has a fix for a similar issue, our docs might still be referring to 4.4.0.

**Comment by wakeldakel** [comment]
Indeed, fixed with 4.4.1!

Thanks so much for the super quick responses.
